contracts/finance/andromeda-validator-staking/src/contract.rs

```diff
@@ -305,3 +310,37 @@ fn query_unstaked_tokens(deps: Deps) -> Result<Vec<UnstakingTokens>, ContractErr
     }
     Ok(res)
 }
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn reply(deps: DepsMut, _env: Env, msg: Reply) -> Result<Response, ContractError> {
+    if msg.result.is_err() {
+        return Err(ContractError::Std(StdError::generic_err(
+            msg.result.unwrap_err(),
+        )));
+    }
+    match ReplyId::from_repr(msg.id) {
+        Some(ReplyId::ValidatorUnstake) => on_validator_unstake(deps, msg),
+        _ => Ok(Response::default()),
+    }
+}
+
+pub fn on_validator_unstake(deps: DepsMut, msg: Reply) -> Result<Response, ContractError> {
+    let attributes = &msg.result.unwrap().events[0].attributes;
```

Comment:
Are we sure that `events` can't be an empty vector? If not we could use `events.first()` instead to cover that case.

Reply:
Events can not be an empty vector as it is called from a reply entry point. Only if `undelegate` msg executed successfully, `on_validator_unstake` is called with `msg` which contains an event for the `undelegate` msg

Reply:
Events can not be an empty vector as it is called from a reply entry point. Only if `undelegate` msg executed successfully, `on_validator_unstake` is called with `msg` which contains an event for the `undelegate` msg